java-plugin/src/main/java/fr/greencodeinitiative/java/checks/OptimizeImageVectorsSize.java

```diff
@@ -0,0 +1,161 @@
+package fr.greencodeinitiative.java.checks;
+
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
+import org.sonar.plugins.java.api.tree.*;
+import org.w3c.dom.*;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerConfigurationException;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+import java.io.File;
+import java.io.IOException;
+import java.io.StringReader;
+import java.lang.reflect.Field;
+import java.rmi.UnmarshalException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
+
+@Rule(
+        key = "EC53",
+        name = "Developpement",
+        description = OptimizeImageVectorsSize.MESSAGERULE,
+        priority = Priority.MINOR,
+        tags = {"bug"})
+public class OptimizeImageVectorsSize extends IssuableSubscriptionVisitor {
+    protected final String SVG_BEGIN = "<svg";
+    protected final String[] SVG_END = new String[]{"</svg", "</ svg>"};
+    protected final List<String> SUPERFLUOUS_ATTRIBUTES = Arrays.asList("xmlns:dc", "xmlns:cc", "xmlns:rdv", "xmlns:svg", "xmlns", "xmlns:sodipodi", "xmlns:inkscape", "inkscape:version", "inkscape:label", "inkscape:groupmode", "sodipodi:docname");
```

Comment:
please could you tell us where do you find this list of superfluous attributes ? I'm curious :p 
If we are ok with you, we could report it to other language rule implementation

Reply:
Hello @dedece35 
I extracted the superfluous attributes from SVGGO demo page, which provides an original & compressed svg file that I compared:
compressed: blob:https://jakearchibald.github.io/09bc7d0d-2148-470d-9a25-86355304127f
original: blob:https://jakearchibald.github.io/b11f1eec-fd24-4bf2-95f2-127555093714
It's not a perfect list !

You can also define custom attributes in svg, e.g: version='1.2'
These attributes are superfluous if they're not used in the rest of the svg.